### ISSUE TITLE
Memory leak in from_string

### ISSUE THREAD

**Issue by ldanielburr** [freach/udatetime#24]
udatetime version 0.13 leaks memory when calling from_string.

Example (Python 2.7):

```python
import udatetime

s = u"2008-09-08T22:47:31-07:00"
while True:
    t = udatetime.from_string(s)
```
The pure python version of from_string does not leak, only the C-extension (from_rfc3339_string) does.

**Comment by elimossrosenberg** [comment]
Yep,
The strdup call on line 124 of rfc3339.c should be free-d similar to the way to the way that the strdup call on line 219 is free-d.

Otherwise, every invocation appears to leak memory

**Comment by freach** [comment]
Thank you @ldanielburr and @elimossrosenberg 
Issue is fixed and fix will be released in 0.0.15
